File: Sem1/ProjetoIntegrador/exerciciosPyton/L01 - pg25_26Exercicio7/L01B.py
#// b)    Ler  uma  temperatura  em  graus  Fahrenheit  e  apresentá-la  convertida  em  graus  Celsius.
#//      A  fórmula  de conversão é C ← (F - 32) * (5/9) , sendo F a temperatura em Fahrenheit e C a temperatura em Celsius.
import time
celcius = 0
validador = True
def verificacao_valor(valor):
    if (valor < (-469.57)):
        valor = True
    else:
        valor = False
    return valor

while validador:
    try:
        fahrenheit = float(input('\ndigite o valor do fahrenheit: ').replace(',','.'))
        if (validador and (fahrenheit < -469.57)):
                    print("\nvalor inválido\n")
                    time.sleep(2)
    except:
        print("valor invalido, digite novamente")
        time.sleep(2)
    

celcius = ((fahrenheit - 32) * (5/9))
resposta = ("O valor "+str(celcius)+"ºC é "+str(fahrenheit)+" em Fahrenheit")

print(resposta)

File: Sem1/ProjetoIntegrador/exercicios/src/tarefaI/estudoDirigidoAlgoritimo/pg25_26Exercicio7/L01B.java
package tarefaI.estudoDirigidoAlgoritimo.pg25_26Exercicio7;

// b)   Ler  uma  temperatura  em  graus  Fahrenheit  e
//      apresentá-la  convertida  em  graus  Celsius.
//      A  fórmula  de conversão é C ← (F - 32) * (5/9) ,
//      sendo F a temperatura em Fahrenheit e C a temperatura em Celsius.

import javax.swing.*;

public class L01B {
    public static void main() {
        double f, c;
        String result;

        c = Double.parseDouble(
                JOptionPane.showInputDialog(
                        "digite  temperatura em fahrenheit:").replace(",","."));

       f = conversionCelcius(c);
       result = String.format("O valor de "+String.format("%.2f",c) +" ºC"+
                       "\nem fahrenheit é: "+String.format("%.2f",f)+" F");

       System.out.println(result);
        JOptionPane.showMessageDialog(null,result);
    }
    public static double conversionCelcius(double fahrenheit) {
        double c = 0;
        c = Double.valueOf(((fahrenheit - 32)*9) / 5);
        return c;
    }
}
